lib/pbench/server/api/resources/users_api.py

```diff
@@ -510,25 +482,28 @@ def delete(self, username):
         # TODO: Check if the user has the right privileges
         if not verified and not user.is_admin():
             self.logger.warning(
-                "Username retrieved from the auth token {} is different from the username provided",
-                self.auth.auth.current_user().id,
+                "User {} is not authorized to delete user {}.", user.username, username,
             )
-            abort(403, message="Authentication token does not belong to the user")
+            abort(403, message="Not authorized to delete user")
 
         try:
-            if not User.delete(username):
-                self.logger.warning("Admin attempted to delete admin user")
+            user = User.query(username=username)
+            # Do not delete if the user is admin
+            if not user.is_admin():
+                User.delete(username)
```

Comment:
I think Dave raised this before, but seeing the code written this way, I'm impelled to ask again:  you really want `delete()` to be a `staticmethod` instead of an instance method?  Given that the code here now goes to the trouble of fetching the instance, it would be more efficient to invoke the deletion on the instance than to delete it by username and have the delete method do its own fetch.

Reply:
Yeah I see that we have to query the same user twice but If delete is an instance method, we would have to do something like 
```
    def delete(self):
        try:
            Database.db_session.delete(self)
            Database.db_session.commit()
        except Exception:
            Database.db_session.rollback()
            raise
```
but then we would be deleting an instance while the control is still in that instance itself. This is causing an `IntegrityError` on `active_tokens` table where it says ` NOT NULL constraint failed on active_tokens.user_id` so doing this is only deleting the user and not the associated active_tokens

Reply:
Ugh.  Fair enough.  (Thanks for trying it!)